gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/common/AbstractTransportServiceStubClassComposer.java

```diff
@@ -1233,12 +1251,200 @@ protected TypeNode getTransportOperationsStubType(Service service) {
     return transportOpeationsStubType;
   }
 
+  protected static LambdaExpr createRequestMutatorClassInstance(
+      Method method, ImmutableMap<String, Message> messageTypes) {
+    List<Statement> bodyStatements = new ArrayList<>();
+    VariableExpr requestVarExpr = createRequestVarExpr(method);
+
+    Reference requestBuilderRef =
+        VaporReference.builder()
+            .setEnclosingClassNames(method.inputType().reference().name())
+            .setName("Builder")
+            .setPakkage(method.inputType().reference().pakkage())
+            .build();
+
+    TypeNode requestBuilderType = TypeNode.withReference(requestBuilderRef);
+
+    VariableExpr requestBuilderVarExpr =
+        VariableExpr.builder()
+            .setVariable(
+                Variable.builder().setName("requestBuilder").setType(requestBuilderType).build())
+            .setIsDecl(true)
+            .build();
+
+    MethodInvocationExpr setRequestBuilderInvocationExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(requestVarExpr)
+            .setMethodName("toBuilder")
+            .setReturnType(requestBuilderType)
+            .build();
+
+    Expr requestBuilderExpr =
+        AssignmentExpr.builder()
+            .setVariableExpr(requestBuilderVarExpr)
+            .setValueExpr(setRequestBuilderInvocationExpr)
+            .build();
+
+    bodyStatements.add(ExprStatement.withExpr(requestBuilderExpr));
+
+    VariableExpr returnBuilderVarExpr =
+        VariableExpr.builder()
+            .setVariable(
+                Variable.builder().setName("requestBuilder").setType(requestBuilderType).build())
+            .setIsDecl(false)
+            .build();
+
+    MethodInvocationExpr.Builder returnExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(returnBuilderVarExpr)
+            .setMethodName("build");
+
+    createRequestMutatorBody(method, messageTypes, bodyStatements, returnBuilderVarExpr);
+
+    return LambdaExpr.builder()
+        .setArguments(requestVarExpr.toBuilder().setIsDecl(true).build())
+        .setBody(bodyStatements)
+        .setReturnExpr(returnExpr.build())
+        .build();
+  }
+
+  @VisibleForTesting
+  static List<Statement> createRequestMutatorBody(
+      Method method,
+      ImmutableMap<String, Message> messageTypes,
+      List<Statement> bodyStatements,
+      VariableExpr returnBuilderVarExpr) {
+
+    if (!shouldGenerateRequestMutator(method, messageTypes)) {
+      return bodyStatements;
+    }
+    Message methodRequestMessage = messageTypes.get(method.inputType().reference().fullName());
+    method.autoPopulatedFields().stream()
+        // Map each field name to its corresponding Field object, if present
+        .map(
+            fieldName ->
+                methodRequestMessage.fields().stream()
+                    .filter(field -> field.name().equals(fieldName))
+                    .findFirst())
```

Comment:
Also a bit lost on this. Why is matching the field name to the message's field required?

Reply:
So the input `messageTypes` is a map of the message name to the actual `Message` for *all* of the messages parsed from the proto. We need to find the exact `Message` that's associated with the field to be populated.

Reply:
I see. So to try and put this all together:
`method.autoPopulatedFields()` is a list of field names from the service yaml file and we need to match those values with the Fields in the list of parsed messages and check if those can be auto-populated. If that's correct, I think that makes sense.

Follow up question: What if multiple messages have the same field inside? i.e.

```
Message A {
 string request_id;
}
```
and
```
Message B {
 string request_id;
}
```

Reply:
A few lines above, the request message is matched first from the input type of the method  
```
Message methodRequestMessage = messageTypes.get(method.inputType().reference().fullName());
```
it guarantees we only try to find the field from the matching request message.